# ***BenchMark***

In [ ]:
from reporte_metricas import ReporteMetricas

report = ReporteMetricas()
df = report.load()


Loaded 9 report(s)


## ***Validation***

### ***macro avg***

In [2]:
dfval_macro = df.query("Type == 'val' and Class == 'macro avg'").drop(columns=['accuracy', 'log_loss', 'auc']).reset_index(drop=True)
dfval_macro.sort_values(by=['f1-score'], ascending=False, inplace=True)
dfval_macro

Model Type      Class precision recall f1-score support
6           XGBOOST  val  macro avg      0.86  0.811    0.833  5085.0
8  XGBOOST_Balanced  val  macro avg     0.844  0.815    0.829  5085.0
2              LGBM  val  macro avg     0.849  0.803    0.824  5085.0
1               MLP  val  macro avg     0.807  0.769    0.786  5085.0
0                RF  val  macro avg     0.766  0.792    0.778  5085.0
3               SVC  val  macro avg     0.756  0.744     0.75  5085.0
5                RL  val  macro avg      0.72  0.771     0.74  5085.0
7               KNN  val  macro avg     0.736  0.639    0.673  5085.0
4              CART  val  macro avg     0.649  0.657    0.652  5085.0

### ***weighted avg***

In [3]:
dfval_weight = df.query("Type == 'val' and Class == 'weighted avg'").drop(columns=['accuracy', 'log_loss', 'auc']).reset_index(drop=True)
dfval_weight.sort_values(by=['f1-score'], ascending=False, inplace=True)
dfval_weight

Model Type         Class precision recall f1-score support
6           XGBOOST  val  weighted avg     0.891  0.895    0.892  5085.0
8  XGBOOST_Balanced  val  weighted avg     0.886  0.889    0.887  5085.0
2              LGBM  val  weighted avg     0.884  0.888    0.885  5085.0
1               MLP  val  weighted avg     0.859  0.864    0.861  5085.0
0                RF  val  weighted avg     0.857  0.849    0.852  5085.0
3               SVC  val  weighted avg     0.835  0.838    0.836  5085.0
5                RL  val  weighted avg     0.833  0.815    0.822  5085.0
7               KNN  val  weighted avg     0.794  0.809    0.794  5085.0
4              CART  val  weighted avg     0.761  0.754    0.757  5085.0

### ***Global***

In [4]:
df_val_global = df.query("Type == 'val' and Class == 'global'").drop(columns=['precision', 'recall', 'f1-score', 'support']).reset_index(drop=True)
df_val_global.sort_values(by=['auc'], ascending=False, inplace=True)
df_val_global

Model Type   Class accuracy log_loss    auc
6           XGBOOST  val  global    0.895    0.294  0.967
2              LGBM  val  global    0.888    0.303  0.964
8  XGBOOST_Balanced  val  global    0.889    0.306  0.963
1               MLP  val  global    0.864    0.367   0.95
0                RF  val  global    0.849    0.463  0.949
3               SVC  val  global    0.838    0.419  0.934
5                RL  val  global    0.815    0.561  0.926
7               KNN  val  global    0.809    1.883    0.9
4              CART  val  global    0.754    0.752  0.843

## ***Test***

### ***macro avg***

In [5]:
dfval_macro = df.query("Type == 'test' and Class == 'macro avg'").drop(columns=['accuracy', 'log_loss', 'auc']).reset_index(drop=True)
dfval_macro.sort_values(by=['f1-score'], ascending=False, inplace=True)
dfval_macro

Model  Type      Class precision recall f1-score support
8  XGBOOST_Balanced  test  macro avg      0.85  0.806    0.826  6355.0
6           XGBOOST  test  macro avg     0.844  0.808    0.825  6356.0
2              LGBM  test  macro avg     0.849  0.793    0.818  6355.0
1               MLP  test  macro avg     0.814  0.767    0.788  6355.0
0                RF  test  macro avg     0.771  0.789    0.779  6355.0
3               SVC  test  macro avg     0.756  0.753    0.754  6356.0
5                RL  test  macro avg     0.733  0.786    0.754  6356.0
7               KNN  test  macro avg     0.745  0.643    0.678  6356.0
4              CART  test  macro avg     0.649  0.657    0.653  6356.0

### ***weighted avg***

In [6]:
dfval_weight = df.query("Type == 'test' and Class == 'weighted avg'").drop(columns=['accuracy', 'log_loss', 'auc']).reset_index(drop=True)
dfval_weight.sort_values(by=['f1-score'], ascending=False, inplace=True)
dfval_weight

Model  Type         Class precision recall f1-score support
8  XGBOOST_Balanced  test  weighted avg     0.887  0.891    0.888  6355.0
6           XGBOOST  test  weighted avg     0.885  0.888    0.886  6356.0
2              LGBM  test  weighted avg     0.884  0.889    0.885  6355.0
1               MLP  test  weighted avg     0.861  0.866    0.862  6355.0
0                RF  test  weighted avg     0.856   0.85    0.852  6355.0
3               SVC  test  weighted avg     0.834  0.836    0.835  6356.0
5                RL  test  weighted avg      0.84  0.823    0.829  6356.0
7               KNN  test  weighted avg     0.797  0.811    0.795  6356.0
4              CART  test  weighted avg     0.762  0.757     0.76  6356.0

Durante el proceso de ***benchmarking*** se compararon distintos modelos de clasificación, recopilando métricas relevantes tanto en los conjuntos de validación como de prueba. El modelo ***XGBoost sin balanceo*** presentó el ***mejor desempeño global*** en términos de ***F1-score*** en validación, alcanzando valores de ***$0.833$ (macro)*** y ***$0.892$ (weighted)***, superando ligeramente al modelo balanceado con ADASYN.

En el conjunto de prueba, si bien el modelo balanceado mostró una ***mejora marginal*** de ***$+0.001$ a $+0.002$*** en el ***F1-score***, estas diferencias no se consideran significativas desde un punto de vista práctico. Por tanto, se concluye que el balanceo ***no mejoró sustancialmente*** el rendimiento del modelo e incluso podría introducir sesgos a través de las muestras minoritarias generadas artificialmente mediante sobremuestreo.

Adicional a las métricas anteriores, se observó que, aunque el modelo balanceado ofreció una ligera ventaja en la clase 'No pago' (***$+0.01$***), ***no mejoró el rendimiento*** en las clases minoritarias ('Pago parcial' y 'Pago completo'), donde el ***F1-score*** fue prácticamente equivalente. Esto refuerza la afirmación de que el balanceo no ofreció beneficios sustanciales y, además, por la naturaleza de la actividad de la organización, esta mejora marginal en la clase mayoritaria no representa ningún beneficio, además de complejizar el modelo.

Otro punto a destacar es la ***estabilidad del modelo XGBoost*** a través de ambos conjuntos, manteniendo diferencias mínimas entre validación y test, lo que indica una ***buena capacidad de generalización***. Desde una perspectiva práctica, el uso de sobremuestreo sintético fue descartado al no aportar mejoras relevantes, evitando así la posible introducción de ruido artificial que afecte la interpretabilidad y robustez del modelo.

En cuanto a la comparación con otros tipos de modelos, se observa una ***superioridad sistemática*** de los modelos basados en ***gradiente***, como ***XGBoost*** y ***LightGBM***, frente a enfoques clásicos como ***árboles de decisión***, ***regresión logística***, ***SVC*** y ***KNN***, e incluso respecto a redes neuronales como el ***MLP***. Esta ventaja se refleja tanto en las métricas promedio (**macro** y **weighted**) como en la consistencia entre los conjuntos de validación y prueba. Además, este comportamiento es coherente con lo reportado en la literatura reciente, donde los modelos de *boosting* han demostrado una excelente capacidad para manejar relaciones no lineales, conjuntos de datos desbalanceados y tareas de clasificación complejas en el ámbito financiero y de riesgo crediticio. En el entorno industrial, su uso también se ha consolidado debido a su ***buen desempeño, velocidad, interpretabilidad y facilidad de ajuste fino***.

Finalmente, el modelo seleccionado como ***ganador*** en el proceso de ***benchmarking*** fue el ***XGBoost sin balanceo***, priorizando tanto su rendimiento técnico como su adaptabilidad a entornos productivos. Sobre este modelo se construyó el sistema de explicación local con LIME y la integración con el motor de inferencia difusa, descritos en las siguientes secciones.